test/benchmark/bench_linux_and_aster.sh

```diff
@@ -61,8 +62,8 @@ run_benchmark() {
     if [ ! -f "${LINUX_KERNEL}" ]; then
         echo "Downloading the Linux kernel image..."
         mkdir -p "${KERNEL_DIR}"
-        curl -L -o "${LINUX_KERNEL}" \
-            -H "Accept: application/vnd.github.v3.raw" \
+        wget --delete-after --header="Accept: application/vnd.github.v3.raw" \
```

Comment:
From wget man page:

> --delete-after
> This option tells Wget to delete every single file it downloads, after having done so.

It seems that you have misunderstood this `--delete-after` option.

Reply:
I think this explanation may lead to misunderstanding. If you run `wget --delete-after -O mongoose.c "https://raw.githubusercontent.com/cesanta/mongoose/7.13/mongoose.c"`, you will see that `mongoose.c.tmp` is downloaded and moved to `mongoose.c` once succeed. 
![image](https://github.com/asterinas/asterinas/assets/24353687/98c226cc-1bf1-4ffa-916b-c9edd302c59f)

Reply:
No, the command `wget --delete-after -O <output_file> <url>` does not work as you intended. 

```
--delete-after
This option tells Wget to delete every single file it downloads, after having done so.
It is useful for pre-fetching popular pages through a proxy, e.g.:
```

The description above has clearly stated that the downloaded file will be deleted always even when the download is successful. Why do we want to _delete_ the file even when the file is downloaded _successfully_?

You are right that `wget --delete-after -O <output_file> <url>` _can_ download the remote resource as `<output_file>`. But that is NOT the point. We can simply use `wget <url>` to download the file, without any arguments. The advanced form of `wget --delete-after -O <output_file> <url>` is supposed to satisfy the following conditions:

1. If the whole file is downloaded, then the file can be found locally;
2. If the file is not downloaded at all or downloaded partially, then the file should not be created.

But `wget --delete-after -O <output_file> <url>` does not work this way.

```
-O file
Use of -O is not intended to mean simply "use the name file instead of the one in the URL ;" rather, it is analogous to shell redirection: wget -O file http://foo is intended to work like wget -O - http://foo > file; file will be truncated immediately, and all downloaded content will be written there.
```

The description above is pretty clear: it works like a redirection. So before the download begins, wget will create a new file as the destination of shell redirection. As the content is downloaded, wget will write the content into both `mongoose.c.tmp` and `mongoose.c`. When the download is complete, the `mongoose.c.tmp` will be deleted, leaving only `mongoose.c`.

The explanation above is supported by experiments. I used CTRL+C to stop wget while it was downloading the remote file. Guess what? An empty `mongoose.c` file remained---as expected. 

<img width="1228" alt="image" src="https://github.com/asterinas/asterinas/assets/568208/d2b3e2a4-03dc-48c5-95b4-426f494aff4a">

Reply:
Thanks for the detailed explanation, @tatetian. It's clear that the --delete-after option combined with -O <output_file> doesn't achieve the desired behavior. The intended use case seems to be more about temporary downloads rather than preserving the file post-download.

For our specific needs, it looks like we'll need to explore alternative approaches or scripts to ensure that the file is only retained if the download is complete even CTRL+C is pressed. Therefore, I think using `TEMP_FILE` is the simplest way.  We download them to `/tmp` and use `wget --continue` to resume transmission after a break.
```makefile
$(MONGOOSE_FILES): | $(MONGOOSE_DIR)
	@if [ ! -d "/tmp" ]; then \
		sudo mkdir /tmp && \
		sudo chmod 1777 /tmp || { \
			echo "Error: Failed to create /tmp directory."; \
			exit 1; \
		}; \
	fi && \
	TARGET_FILE=$(notdir $@) && \
	TEMP_FILE="/tmp/$$TARGET_FILE" && \
	wget --continue -O "$$TEMP_FILE" "https://raw.githubusercontent.com/cesanta/mongoose/7.13/$$TARGET_FILE" && \
	mv "$$TEMP_FILE" "$@" 
```

Thanks again for pointing out the nuances of the command. It's always helpful to get a deeper understanding of how these tools work under the hood.